database/gorm/query.go

```diff
@@ -619,21 +666,67 @@ func (r *QueryImpl) With(query string, args ...any) ormcontract.Query {
 		case func(ormcontract.Query) ormcontract.Query:
 			newArgs := []any{
 				func(tx *gormio.DB) *gormio.DB {
-					query := arg(NewQueryWithWithoutEvents(tx, r.withoutEvents))
+					query := arg(NewQueryWithWithoutEvents(tx, r.withoutEvents, r.config))
 
 					return query.(*QueryImpl).instance
 				},
 			}
 
 			tx := r.instance.Preload(query, newArgs...)
 
-			return NewQueryWithWithoutEvents(tx, r.withoutEvents)
+			return NewQueryWithWithoutEvents(tx, r.withoutEvents, r.config)
 		}
 	}
 
 	tx := r.instance.Preload(query, args...)
 
-	return NewQueryWithWithoutEvents(tx, r.withoutEvents)
+	return NewQueryWithWithoutEvents(tx, r.withoutEvents, r.config)
+}
+
+func driver2gorm(driver string) string {
+	switch driver {
+	case "mysql":
+		return "mysql"
+	case "postgresql":
+		return "postgres"
+	case "sqlite":
+		return "sqlite"
+	case "sqlserver":
+		return "sqlserver"
+	}
+	return ""
+}
+
+func (r *QueryImpl) refreshConnection(value any) error {
+	model, ok := value.(ormcontract.ConnectionModel)
+	if !ok {
+		return nil
+	}
+	conn := model.Connection()
+	if conn == "" {
+		return fmt.Errorf("connection shouldn't be empty")
+	}
+	driver := driver2gorm(r.config.GetString(fmt.Sprintf("database.connections.%s.driver", conn)))
+	if driver == "" {
+		return fmt.Errorf("connection %s driver is not supported", conn)
+	}
```

Comment:
S: Add some test cases for these three situations.

Reply:
Okay